src/Microsoft.Graph.Core/Exceptions/Error.cs

```diff
@@ -4,47 +4,43 @@
 
 namespace Microsoft.Graph
 {
+    using Microsoft.Kiota.Abstractions.Serialization;
```

Comment:
I don't believe these are meant to conflict with the one used in the service library as using the service library will use the generated errors from the models. 
However, they exist to provide a way to provide strongly typed errors when using the core only features such as FileUploads and Batching which provide ability to leverage them in customized response handlers. 
Let me know what you think.

Reply:
To avoid this sort of duplication, what I've instructed other languages to do is to create a base (abstract) implementation in core that accepts the error mappings as a parameter of their constructors (like we have in the generated executor methods) and then use a derived type in each service lib to provide the mapping. Do you think this is something we can address before GA?

Reply:
Cleaned up the error handling in https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/pull/607